tests/test_host_check.py

```diff
@@ -3267,6 +3167,167 @@ def test_warn_igb_uio_enabled(self, capsys):
         assert result is CheckState.NEUTRAL
 
 
+class TestPCIDevices(_CheckTestBase):
+    """Tests for the PCI devices check."""
+
+    check_group = "xrd-vrouter"
+    check_name = "PCI devices"
+    deps = ["IOMMU"]
+    cmds = ["lshw -businfo -c network"]
+    files = []
+
+    def test_success_vfio_iommu(self, capsys):
+        """Test the success case for vfio with IOMMU enabled."""
+        lshw_output = """\
+Bus info          Device      Class      Description
+====================================================
+pci@0000:00:00.0  device1     network    82540EM Gigabit Ethernet Controller
+pci@0000:00:1f.2  device4     network    Ethernet interface
+pci@0000:00:02.0  device3     network    82540EM Gigabit Ethernet Controller
+pci@0000:00:01.0  device2     network    Ethernet interface
+                  docker0     network    Ethernet interface
+
+
+        """
+        iommu_devices = [
+            "0000:00:00.0",
+            "0000:00:01.0",
+            "0000:00:02.0",
+            "0000:00:1f.0",
+            "0000:00:1f.3",
+        ]
+        with mock.patch("glob.glob", return_value=iommu_devices):
+            result, output = self.perform_check(
+                capsys,
+                cmd_effects=[lshw_output],
+                read_effects="N",
+            )
+        assert textwrap.dedent(output) == textwrap.dedent(
+            f"""\
+            PASS -- PCI devices
+                    The following PCI device(s) are available:
+                    device1 (0000:00:00.0), device2 (0000:00:01.0), device3 (0000:00:02.0)
+            """
+        )
+        assert result is CheckState.SUCCESS
+
+    def test_success_igb_uio(self, capsys):
+        """Test the success case for igb_uio enabled."""
+        lshw_output = """\
+Bus info          Device      Class      Description
+====================================================
+pci@0000:00:00.0  device1     network    82540EM Gigabit Ethernet Controller
+pci@0000:00:1f.2  device4     network    Ethernet interface
+pci@0000:00:02.0  device3     network    82540EM Gigabit Ethernet Controller
+pci@0000:00:01.0  device2     network    Ethernet interface
+                  docker0     network    Ethernet interface
+
+
+        """
+        result, output = self.perform_check(
```

Comment:
I'm not sure how this works without mocking `glob.glob`?!

Reply:
Yeah, `glob.glob` should be mocked here (I guess it worked anyway due to the exception catching)